src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java

```diff
@@ -313,8 +313,11 @@ public AbstractGroup resultConverter(ButtonType button) {
                             groupName,
                             groupHierarchySelectedProperty.getValue(),
                             FieldFactory.parseField(autoGroupKeywordsFieldProperty.getValue().trim()),
-                            autoGroupKeywordsDelimiterProperty.getValue().charAt(0),
-                            autoGroupKeywordsHierarchicalDelimiterProperty.getValue().charAt(0));
+                            autoGroupKeywordsDelimiterProperty.getValue().isEmpty() ?
```

Comment:
I find this code very hard to read. Could you extract if statements generating the value into a variable and then use the variable?